tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
+    bluechi_is_online: BluechiIsOnline,
+    future: ResultFuture,
+    description: str,
+    wait_time: int = 0,
+    expected_result: int = 0,
+):
+    result, output = bluechi_is_online.run(
+        description,
+        f"agent --wait {wait_time}",
+        check_result=False,
+        expected_result=expected_result,
+    )
+    future.result = result
+    future.output = output
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Start agent and verify --wait option returns 0 immediately
+    node_foo.systemctl.start_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+    LOGGER.debug("Starting test number 1 - agent should be online.")
+
+    result_future_online = ResultFuture()
+    check_thread_online = threading.Thread(
+        target=check_agent,
+        args=(
+            node_foo.bluechi_is_online,
+            result_future_online,
+            "Checking if agent is online",
+            0,
+            0,
+        ),
+    )
+    check_thread_online.start()
+    check_thread_online.join()
```

Comment:
Yes, since `bluechi-is-online` should exit immediately and we don't need to do anything in parallel, I'd just wrap this in a timeout - no need for the complexity of threads here. Something like this:
```python
wait_time=0
result = False
start_time = time.time()
with Timeout(wait_time+1, "bluechi-is-online didn't return immediately"):
    result = node_foo.bluechi_is_online(wait_time)
assert result, f"Expected bluechi-is-online with --wait={wait_time} on node {NODE_FOO} to exit true"
assert start_time - time.time() < 1, f"Expected less than 1 second for bluechi-is-online to exit"
```